testsuite/cluster-test/src/tx_emitter.rs

```diff
@@ -649,6 +653,23 @@ fn invalid_tx(
         InvalidTxType::ChainId => {
             gen_transfer_txn_request(sender, receiver, SEND_AMOUNT, ChainId::new(255), gas_price)
         }
+        InvalidTxType::Duplication => {
+            // if this is the first tx, default to generate invalid tx with wrong chain id
+            // otherwise, make a duplication of an exist valid tx
+            if reqs.len() == 0 {
+                gen_transfer_txn_request(
+                    sender,
+                    receiver,
+                    SEND_AMOUNT,
+                    ChainId::new(255),
+                    gas_price,
+                )
+            } else {
+                let mut rng = rand::thread_rng();
```

Comment:
Should you be using the RNG that is initialized above? This would help reduce the sources of randomness if we wanted to seed the whole operation at some point.